extension/js/content_scripts/webmail/gmail-element-replacer.ts

```diff
@@ -746,4 +749,47 @@ export class GmailElementReplacer implements WebmailElementReplacer {
     }
   }
 
+  private renderLocalDrafts = async () => {
+    if (window.location.hash === '#drafts') {
+      const storage = await GlobalStore.get(['local_drafts']);
+      if (!storage.local_drafts) {
+        return;
+      }
+      const offlineComposeDrafts: Dict<number> = {};
+      for (const draftId in storage.local_drafts) {
+        if (draftId.startsWith('local-draft-compose-')) {
+          offlineComposeDrafts[draftId] = parseInt(storage.local_drafts[draftId].id);
+        }
+      }
+      let offlineDraftsContainer = $(this.sel.draftsList).find('#fc_offline_drafts');
+      if (!offlineDraftsContainer.length) {
+        offlineDraftsContainer = $('<div id="fc_offline_drafts"></div>');
+        $(this.sel.draftsList).append(offlineDraftsContainer); // xss-safe-value
+      }
+      if (Object.keys(offlineComposeDrafts).length) {
+        const draftsToRender: JQueryEl[] = [];
```

Comment:
why is this array `draftsToRender` keyed differently than `storage.local_drafts`? Is this only to call `sort()` without a sorting function/lambda parameter?
`local-draft-compose-HMhqrZKsUo`  should be a good key, and its uniqueness is in line with the storage implementation. I see the click  handler operates with storage keys. 

The first thing that perplexed me is why `id` inside a draft object differs from the outer `draftId` in storage.
Is the only purpose to make it sortable? I guess this is unnecessary.
Also, it is named `id` as if its meaning is unknown (opaque) to the outside modules, and then on line 771 you apply the knowledge that it is actually a timestamp to format it with `new Date(...).toLocaleString()`.
This knowledge (requirement) should be explicitly reflected in the property name.
So I suggest for drafts to not have `id` field (or matching with the outside id, if required).
Have `timestamp` field instead. It is usually inique but not really required to be.
Do not create any dictionaries with `timestamp` as key. This is unnecessary complication and uniqueness requirement.

Reply:
Thank you for the advice @rrrooommmaaa, I followed it and the code is cleaner now 👍